packages/opentelemetry-core/src/utils/lodash.merge.ts

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+/**
+ * based on lodash.merge
+ * @TODO - should I add here more info ?
```

Comment:
i think we need to explain why we vendored it yeah

Reply:
I agree but it doesn't need to be a huge explanation. Something like "copied from lodash.merge in order to support esm builds without esModuleInterop" should be sufficient.

Reply:
will add that, was rather curious with regards to any license / etc. to be sure we don't miss something any disclaimer etc. or this info is enough ?

Reply:
i think its enough. maybe mention which license lodash uses?